### PR TITLE
subconfigs

### LINKED ISSUE #29: create subconfigs

### PR DESCRIPTION
**Pull request by KilianFt**
closes #29

### CODE REVIEW

**Review by manuel-delverme — COMMENTED** [review]
Great idea.
Do you think we can pass only the subconfig to each step?

File: finetune_user_model.py [hunk 1]
```diff
@@ -98,7 +91,8 @@ def main(model, user_hash, config: configs.BaseConfig):
     }
 
     if sys.gettrace() is not None:
-        main(base_configuration)
+        raise NotImplementedError
     else:
+        raise NotImplementedError
```
Comment (manuel-delverme) [inline finetune_user_model.py:96]: :D

File: main.py [hunk 2]
```diff
@@ -20,6 +20,7 @@ def main():
         experiment_config = configs.SmokeConfig()
     else:
         experiment_config = configs.BaseConfig()
+    experiment_config = configs.SmokeConfig()
```
Comment (manuel-delverme) [inline main.py:23]: remember to remove this before merging

**Comment by KilianFt** [comment]
> Great idea. Do you think we can pass only the subconfig to each step?

Rebase was kinda messy, I had an old main locally but should be fine now.
There are these 1-2 times where we use a general parameter and it would be kinda messy to copy them. Don't think it matters much to send all right?
Maybe we could have the diskcache only consider the specific config for each step

**Pull request merged**